### PR TITLE
TW-203: pull to refresh on tokens page

### PR DESCRIPTION
**Pull request by letier3110**
feat: update balances once a block

### CODE REVIEW

File: src/hooks/block-subscription/block-subscription.hook.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  protocol: '',
+  chain_id: '',
+  hash: '',
+  header: {
+    level: 0,
+    timestamp: ''
+  }
+};
+
+export const useBlockSubscription = (webSocketUrl: string) => {
```
Comment (LendiHop) [inline src/hooks/block-subscription/block-subscription.hook.ts:18]: rename file to use-block-subscription.hook

File: src/hooks/block-subscription/block-subscription.hook.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+export const EMPTY_BLOCK: BlockInterface = {
+  protocol: '',
+  chain_id: '',
+  hash: '',
+  header: {
+    level: 0,
+    timestamp: ''
+  }
+};
```
Comment (LendiHop) [inline src/hooks/block-subscription/block-subscription.hook.ts:16]: mb extract to separate file?

File: src/screens/wallet/wallet.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import { HeaderCardActionButtons } from '../../components/header-card-action-but
 import { HeaderCard } from '../../components/header-card/header-card';
 import { IconNameEnum } from '../../components/icon/icon-name.enum';
 import { TouchableIcon } from '../../components/icon/touchable-icon/touchable-icon';
+// import { ScreenContainer } from '../../components/screen-container/screen-container';
```
Comment (LendiHop) [inline src/screens/wallet/wallet.tsx:11]: rm

File: src/hooks/block-subscription/block-subscription.hook.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  const onRefresh = () => {
+    setIsRefreshing(true);
+    refreshControlRef.current = Math.random();
+  };
+
+  useEffect(() => {
+    webSocketRef.current = new WebSocket(webSocketUrl);
+
+    webSocketRef.current.onerror = (errorEvent: Event) => {
+      console.log(errorEvent);
```
Comment (LendiHop) [inline src/hooks/block-subscription/block-subscription.hook.ts:35]: rm

File: src/navigator/main-stack.tsx [hunk 5]
```diff
@@ -82,7 +86,11 @@ export const MainStackScreen = () => {
     dispatch(loadExchangeRates.submit());
   };
 
-  useAuthorisedTimerEffect(initDataLoading, DATA_REFRESH_INTERVAL, [selectedAccount.publicKeyHash, selectedRpcUrl]);
+  useTimerEffect(initDataLoading, DATA_REFRESH_INTERVAL, [
```
Comment (unixvb) [inline src/navigator/main-stack.tsx:89]: why did you change `useAuthorisedTimerEffect` to `useTimerEffect`?

File: src/navigator/main-stack.tsx [hunk 6]
```diff
@@ -82,7 +86,11 @@ export const MainStackScreen = () => {
...
+    blockSubscription,
```
Comment (unixvb) [inline src/navigator/main-stack.tsx:90]: Maybe `blockSubscription.block.header.level` instead of `blockSubscription`?

File: src/navigator/main-stack.tsx (hunk 5 above)
Comment (LendiHop) [inline src/navigator/main-stack.tsx:89]: why not useAuthorisedTimerEffect?

File: src/hooks/block-subscription/block-subscription.hook.ts (hunk 1 above)
Comment (unixvb) [inline src/hooks/block-subscription/block-subscription.hook.ts:18]: `webSocketUrl` parameter would not be reused.  
just create a local module variable 
```
const BLOCK_UPDATE_API_URL = `${TEZOS_DEXES_API_URL}/block`;
```

and reuse it

File: src/screens/wallet/wallet.tsx [hunk 7]
```diff
@@ -27,37 +28,49 @@ export const Wallet = () => {
...
-        <Divider size={formatSize(16)} />
+          <Divider size={formatSize(16)} />
 
-        <CollectiblesHomeSwipeButton />
-      </HeaderCard>
+          <CollectiblesHomeSwipeButton />
+        </HeaderCard>
 
-      <TokenList />
+        <TokenList />
```
Comment (unixvb) [inline src/screens/wallet/wallet.tsx:72]: I think you have received error in console that `FlatList` in inside `ScrollView`

File: src/screens/wallet/wallet.tsx [hunk 8]
```diff
@@ -27,37 +28,49 @@ export const Wallet = () => {
...
   const visibleAccounts = useVisibleAccountsListSelector();
   const tezosToken = useSelectedAccountTezosTokenSelector();
 
   usePageAnalytic(ScreensEnum.Wallet);
 
+  const handleRefresh = () => {
+    setRefreshing(true);
+    setTimeout(() => {
+      setRefreshing(false);
+    }, 3000);
```
Comment (unixvb) [inline src/screens/wallet/wallet.tsx:43]: make this value random, please. from 1500 to 3000 for example

File: src/navigator/main-stack.tsx (hunk 5 above)
Comment (letier3110) [reply src/navigator/main-stack.tsx:89]: because there was simple settimeout change at beginning, but it was not firing events more than once

File: src/screens/wallet/wallet.tsx [hunk 9]
```diff
@@ -27,37 +27,54 @@ export const Wallet = () => {
...
-      </HeaderCard>
+              <Divider size={formatSize(16)} />
 
-      <TokenList />
+              <CollectiblesHomeSwipeButton />
+            </HeaderCard>
+            <TokenList />
+          </>
+        )}
+        refreshControl={<RefreshControl refreshing={isRefreshing} onRefresh={handleRefresh} />}
```
Comment (LendiHop) [inline src/screens/wallet/wallet.tsx:76]: mb move this logic into TokenList component?

**Review by LendiHop — APPROVED** [review]

**Review by unixvb — APPROVED** [review]

File: src/components/activity-groups-list/activity-groups-list.tsx [hunk 10]
```diff
@@ -41,6 +43,13 @@ export const ActivityGroupsList: FC<Props> = ({ activityGroups, handleUpdate = e
     return result;
   }, [activityGroups]);
 
+  const handleRefresh = () => {
```
Comment (LendiHop) [inline src/components/activity-groups-list/activity-groups-list.tsx:46]: extract this handler to a separate file and reuse it here and in TokenList component

**Review by LendiHop — APPROVED** [review]

File: src/hooks/use-refresh.hook.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react';
+
+export const useRefresh = () => {
+  const [isRefreshing, setRefreshing] = useState(false);
+
+  const handleRefresh = () => {
+    setRefreshing(true);
+    setTimeout(() => {
```
Comment (unixvb) [inline src/hooks/use-refresh.hook.ts:8]: clear timer on unmount (look like `useActiveTimer` used for this)

**Review by LendiHop — DISMISSED** [review]

File: src/hooks/use-refresh.hook.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    }, Math.random() * 2000 + 2000);
+  };
+
+  useEffect(() => {
+    return () => {
+      if (isDefined(timerRef.current)) {
+        clearTimeout(timerRef.current);
+      }
+    };
+  }, []);
```
Comment (unixvb) [inline src/hooks/use-refresh.hook.ts:23]: why don;t you reuse `useActiveTimer`??

**Review by unixvb — APPROVED** [review]

**Review by oryhivskiy — APPROVED** [review]
